public/app/plugins/panel/canvas/CanvasContextMenu.tsx

```diff
@@ -99,6 +99,41 @@ export const CanvasContextMenu = ({ scene }: Props) => {
       return null;
     };
 
+    const typeOptions = getElementTypes(scene.shouldShowAdvancedTypes).options;
+
+    const getTypeOptionsSubmenu = () => {
+      const submenuItems: Array<React.ReactElement<MenuItemProps<any>, string | React.JSXElementConstructor<any>>> = [];
+      typeOptions.map((option) => {
+        submenuItems.push(
+          <MenuItem key={option.value} label={option.label!} onClick={() => onAddItem(option, rootLayer)} />
+        );
+      });
+
+      return submenuItems;
+    };
+
+    const addItemMenuItem = !scene.isPanelEditing && (
+      <MenuItem
+        label="Add item"
+        className={styles.menuItem}
+        childItems={getTypeOptionsSubmenu()}
+        customStyle={{ maxHeight: '150px', overflowY: 'auto' }}
```

Comment:
maybe rather than adding generic `customStyle` to MenuItem, we could add `subMenuMaxHeight` and then set overflowY inside the component?

Reply:
we could do that too, but I think it feels more like a limitation. I think a custom style goes a long way long term (you never know when we might decide to have a red submenu..)